tests/RobotFramework/tests/mqtt/disconnect_from_broker.robot

```diff
@@ -0,0 +1,32 @@
+*** Settings ***
+Resource        ../../resources/common.resource
+Library         ThinEdgeIO
+Library         OperatingSystem
+Library         String
+
+Suite Setup     Setup
+Suite Teardown  Get Logs
+
+Test Tags      theme:mqtt
+
+
+*** Test Cases ***
+
+Validate no Error message in journalctl
+    [Documentation]    This Test step is inspired by 'tedge mqtt cli commands do not disconnect cleanly from MQTT broker #2922' bug
+    ...                extended to check for any possible error that may appear in the journalctl
+    Execute Command    tedge mqtt pub foo bar
+    ${log}    Execute Command    journalctl -u mosquitto -n 30
+    Should Not Contain    ${log}    Error
```

Comment:
Setting up TLS in not so straight forward and the bootstrap.sh script (used in the setup of the device under test) takes care of setting up TLS (if activated)...so doing this in a specific test to either opt-out or opt-in to TLS will definitely complicate the test considerably.

Reply:
> I created the original ticket with TLS included just because I was using TLS on my device. I didn't explicitly test it without TLS, but I assume it would have no bearing on the outcome.

I tested without TLS when fixing the issue. There were no errors reported in mosquitto logs.

Reply:
Ok, then sorry for the confusion. I guess I was just thinking about the assertion which checks for the `DISCONNECT` message and not the absence of the `Error` messages.